res/controllers/Numark-Party-Mix.scripts.js

```diff
@@ -0,0 +1,440 @@
+                    }
+                } else {
+                    engine.setValue(this.group, "start_stop", 1);
+                }
+            }
+        });
+    }
+};
+NumarkPartyMix.ModeSampler.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkPartyMix.ModeEFX = function(deckNumber) {
+    components.ComponentContainer.call(this);
+
+    // this.control = NumarkPartyMix.PadModeControls.EFX;
+
+    var fx = [
+        "[EffectRack1_EffectUnit" + deckNumber + "_Effect1]",
+        "[EffectRack1_EffectUnit" + deckNumber + "_Effect2]",
+        "[EffectRack1_EffectUnit" + deckNumber + "_Effect3]"
+    ];
+
+    this.connections = new components.ComponentContainer();
+    var p = this.connections;
+
+    fx.forEach(function(fx, i) {
+        p[i] = new components.Button({
+            midi: [0x93 + deckNumber, 0x14 + i],
+            group: fx,
+            inKey: "enabled",
+            outKey: "enabled",
+            outConnect: false
+        });
+    })
+    p[3] = new components.Button({
+        midi: [0x93 + deckNumber, 0x17],
+        type: components.Button.prototype.types.toggle,
+        group: "[EffectRack1_EffectUnit" + deckNumber + "]",
+        inKey: "mix_mode",
+        outKey: "mix_mode",
+        output: function() {
+            if (engine.getParameter("[EffectRack1_EffectUnit" + deckNumber + "]", "mix_mode") === 0) {
+                midi.sendShortMsg(0x93 + deckNumber, 0x17, 0x7F); // cue
+            } else {
+                midi.sendShortMsg(0x83 + deckNumber, 0x17, 0x01); // cue
+            }
+        },
+        outConnect: false
+    });
+};
+NumarkPartyMix.ModeEFX.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkPartyMix.Browse = function() {
+    var pressturn = false;
+
+    this.knob = new components.Encoder({
+        input: function(channel, control, value) {
+            var direction;
+            if (pressturn) {
+                if (value > 0x40) {
+                    engine.setParameter("[Library]", "MoveFocusForward", 1);
+                } else {
+                    engine.setParameter("[Library]", "MoveFocusBackward", 1);
+                }
+            } else {
+                direction = (value > 0x40) ? -1 : 1;
+                engine.setParameter("[Library]", "MoveVertical", direction);
+            }
+        }
+    });
+
+    this.knobButton = new components.Button({
+        group: "[Library]",
+        type: components.Button.prototype.types.powerWindow,
+        isLongPressed: false,
+        input: function(channel, control, value, status, group) {
+            pressturn = this.isPress(channel, control, value, status);
+            print("pressturn");
+            print(pressturn);
+            if (pressturn) {
+                this.inToggle(channel, control, value, status, group);
+                this.isLongPressed = false;
+                this.longPressTimer = engine.beginTimer(this.longPressTimeout, function() {
```

Comment:
where is `longPressTimeout` defined?

Reply:
https://github.com/mixxxdj/mixxx/blob/6eb12c02d2e60e9128cd0d20672b489c3394c155/res/controllers/midi-components-0.0.js#L178